platform/ios/PolyPodApp/PolyPod/Authentication.swift

```diff
@@ -35,21 +35,31 @@ class Authentication {
         UserDefaults.standard.set(true, forKey: Authentication.disableCheckKey)
     }
     
-    func setUp(_ completeAction: @escaping (Bool) -> Void) {
-        authenticateLocally(reason: "auth_prompt_set_up") { success in
-            self.authenticated = success
-            UserDefaults.standard.set(true, forKey: Authentication.setUpKey)
+    
+    func setUp(newStatus: Bool, _ completeAction: @escaping (Bool) -> Void) {
+        let reason = isSetUp() ?
+        "re_auth_prompt_set_up": "auth_prompt_set_up"
+
+        authenticateLocally(withReason: reason) { success in
+            if (success) {
+                self.authenticated = newStatus
+                UserDefaults.standard.set(newStatus, forKey: Authentication.setUpKey)
+            }
             completeAction(success)
         }
     }
+            
     
     func authenticate(_ completeAction: @escaping (Bool) -> Void) {
         if isCheckDisabled() || !isSetUp() || authenticated {
             completeAction(true)
             return
         }
         
-        authenticateLocally(reason: "auth_prompt_unlock") { success in
+        let reason = isSetUp() ?
```

Comment:
I don't understand something, above at line 54 we have `!isSetUp()` and then here we check again for `isSetuUp`, which at this point will always be true, which means that user will always see "re_auth_prompt_unlock"?

Reply:
good catch! I think it's a remainant from the previous `authenticate()` and as I was using a second function when merged them into one, this was passed by.
You're correct, in our new flow the first `if` doesn't need the check for setup. 
Thanks!

Reply:
But then, which message should be used "re_auth_prompt_unlock" or "auth_prompt_unlock"? I would say the later.

Reply:
the "Login to your polyPod"?
I'd say both would be correct as the user is not authenticated in this case, but happens to use the app and it's back on the "lock state". Can leave it to 'Login to your polyPod' as it was before till we've better guidelines